<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Изучение-данных-из-файла" data-toc-modified-id="Изучение-данных-из-файла-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Изучение данных из файла</a></span><ul class="toc-item"><li><span><a href="#Изучение-данных-из-файла-hypothesis" data-toc-modified-id="Изучение-данных-из-файла-hypothesis-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Изучение данных из файла hypothesis</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-1.1.1"><span class="toc-item-num">1.1.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Датасет-orders" data-toc-modified-id="Датасет-orders-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Датасет orders</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-1.2.1"><span class="toc-item-num">1.2.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Датасет-visitors" data-toc-modified-id="Датасет-visitors-1.3"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Датасет visitors</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-1.3.1"><span class="toc-item-num">1.3.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-1.4"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Приоритизация-гипотез" data-toc-modified-id="Приоритизация-гипотез-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Приоритизация гипотез</a></span><ul class="toc-item"><li><span><a href="#Применить--фреймворк-ICE-для-приоритизации-гипотез.Отсортировать-по-убыванию-приоритета." data-toc-modified-id="Применить--фреймворк-ICE-для-приоритизации-гипотез.Отсортировать-по-убыванию-приоритета.-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Применить  фреймворк ICE для приоритизации гипотез.Отсортировать по убыванию приоритета.</a></span></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Вывод</a></span></li><li><span><a href="#Применить--фреймворк-RICE-для-приоритизации-гипотез.Отсортировать-по-убыванию-приоритета." data-toc-modified-id="Применить--фреймворк-RICE-для-приоритизации-гипотез.Отсортировать-по-убыванию-приоритета.-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Применить  фреймворк RICE для приоритизации гипотез.Отсортировать по убыванию приоритета.</a></span></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Вывод</a></span></li><li><span><a href="#Указать,-как-изменилась-приоритизация-гипотез-при-применении-RICE-вместо-ICE" data-toc-modified-id="Указать,-как-изменилась-приоритизация-гипотез-при-применении-RICE-вместо-ICE-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Указать, как изменилась приоритизация гипотез при применении RICE вместо ICE</a></span></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-2.6"><span class="toc-item-num">2.6&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Анализ-А/В-теста" data-toc-modified-id="Анализ-А/В-теста-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Анализ А/В теста</a></span><ul class="toc-item"><li><span><a href="#Построить-график-кумулятивной-выручки-по-группам" data-toc-modified-id="Построить-график-кумулятивной-выручки-по-группам-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Построить график кумулятивной выручки по группам</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.1.1"><span class="toc-item-num">3.1.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Построить-график-кумулятивного-среднего-чека-по-группам" data-toc-modified-id="Построить-график-кумулятивного-среднего-чека-по-группам-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Построить график кумулятивного среднего чека по группам</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.2.1"><span class="toc-item-num">3.2.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Построить-график-относительного-изменения-кумулятивного-среднего-чека-группы-В-к-группе-А" data-toc-modified-id="Построить-график-относительного-изменения-кумулятивного-среднего-чека-группы-В-к-группе-А-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Построить график относительного изменения кумулятивного среднего чека группы В к группе А</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.3.1"><span class="toc-item-num">3.3.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Построить-график-кумулятивной-конверсии-по-группам" data-toc-modified-id="Построить-график-кумулятивной-конверсии-по-группам-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Построить график кумулятивной конверсии по группам</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.4.1"><span class="toc-item-num">3.4.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Построить-график-относительного-изменения-кумулятивной-конверсии-группы-В-к-группе-А" data-toc-modified-id="Построить-график-относительного-изменения-кумулятивной-конверсии-группы-В-к-группе-А-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Построить график относительного изменения кумулятивной конверсии группы В к группе А</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.5.1"><span class="toc-item-num">3.5.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Построить-точечный-график-количества-заказов-по-пользователям" data-toc-modified-id="Построить-точечный-график-количества-заказов-по-пользователям-3.6"><span class="toc-item-num">3.6&nbsp;&nbsp;</span>Построить точечный график количества заказов по пользователям</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.6.1"><span class="toc-item-num">3.6.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Посчитать-95-й-и-99-й-перцентили-количества-заказов-на-пользователя.-Выбрать-границу-для-определения-аномальных-заказов" data-toc-modified-id="Посчитать-95-й-и-99-й-перцентили-количества-заказов-на-пользователя.-Выбрать-границу-для-определения-аномальных-заказов-3.7"><span class="toc-item-num">3.7&nbsp;&nbsp;</span>Посчитать 95-й и 99-й перцентили количества заказов на пользователя. Выбрать границу для определения аномальных заказов</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.7.1"><span class="toc-item-num">3.7.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Построить-точечный-график-стоимости-заказов." data-toc-modified-id="Построить-точечный-график-стоимости-заказов.-3.8"><span class="toc-item-num">3.8&nbsp;&nbsp;</span>Построить точечный график стоимости заказов.</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.8.1"><span class="toc-item-num">3.8.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Посчитать-95-й-и-99-й-перцентили-стоимости-заказов.-Выбрать-границу-для-определения-аномальных-заказов" data-toc-modified-id="Посчитать-95-й-и-99-й-перцентили-стоимости-заказов.-Выбрать-границу-для-определения-аномальных-заказов-3.9"><span class="toc-item-num">3.9&nbsp;&nbsp;</span>Посчитать 95-й и 99-й перцентили стоимости заказов. Выбрать границу для определения аномальных заказов</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.9.1"><span class="toc-item-num">3.9.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Посчитать-статистическую-значимость-различий-в-конверсии-между-группами-по-&quot;сырым&quot;-данным." data-toc-modified-id="Посчитать-статистическую-значимость-различий-в-конверсии-между-группами-по-&quot;сырым&quot;-данным.-3.10"><span class="toc-item-num">3.10&nbsp;&nbsp;</span>Посчитать статистическую значимость различий в конверсии между группами по "сырым" данным.</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.10.1"><span class="toc-item-num">3.10.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Посчитать-статистическую-значимость-различий-в-среднем-чеке-заказа-между-группами-по-&quot;сырым&quot;-данным." data-toc-modified-id="Посчитать-статистическую-значимость-различий-в-среднем-чеке-заказа-между-группами-по-&quot;сырым&quot;-данным.-3.11"><span class="toc-item-num">3.11&nbsp;&nbsp;</span>Посчитать статистическую значимость различий в среднем чеке заказа между группами по "сырым" данным.</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.11.1"><span class="toc-item-num">3.11.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Посчитать-статистическую-значимость-различий-в-конверсии-между-группами-по-&quot;очищенным&quot;-данным" data-toc-modified-id="Посчитать-статистическую-значимость-различий-в-конверсии-между-группами-по-&quot;очищенным&quot;-данным-3.12"><span class="toc-item-num">3.12&nbsp;&nbsp;</span>Посчитать статистическую значимость различий в конверсии между группами по "очищенным" данным</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.12.1"><span class="toc-item-num">3.12.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Посчитать-статистическую-значимость-различий-в-среднем-чеке-заказа-между-группами-по-&quot;очищенным&quot;-данным." data-toc-modified-id="Посчитать-статистическую-значимость-различий-в-среднем-чеке-заказа-между-группами-по-&quot;очищенным&quot;-данным.-3.13"><span class="toc-item-num">3.13&nbsp;&nbsp;</span>Посчитать статистическую значимость различий в среднем чеке заказа между группами по "очищенным" данным.</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.13.1"><span class="toc-item-num">3.13.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Принять-решения-по-результатам-теста" data-toc-modified-id="Принять-решения-по-результатам-теста-3.14"><span class="toc-item-num">3.14&nbsp;&nbsp;</span>Принять решения по результатам теста</a></span><ul class="toc-item"><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.14.1"><span class="toc-item-num">3.14.1&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-3.15"><span class="toc-item-num">3.15&nbsp;&nbsp;</span>Вывод</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

# Отдел Аналитики крупного интернет магазина.
Вместе с отделом маркетинга подготовлен список гипотез для увеличения выручки.
Цель приоритизировать гипотезы, запустить A/B тест и проанализировать результаты

## Изучение данных из файла

In [1]:
import pandas as pd
from IPython.display import display
import numpy as np
import scipy.stats as sps
import matplotlib.pyplot as plt
import seaborn as sns
import datetime as dt
import warnings
warnings.filterwarnings('ignore')

### Изучение данных из файла hypothesis

In [2]:
hypothesis = pd.read_csv('/datasets/hypothesis.csv', sep = ',')
pd.options.display.max_colwidth = 150  
hypothesis.head(10)

Hypothesis  \
0            Добавить два новых канала привлечения трафика, что позволит привлекать на 30% больше пользователей   
1                                     Запустить собственную службу доставки, что сократит срок доставки заказов   
2  Добавить блоки рекомендаций товаров на сайт интернет магазина, чтобы повысить конверсию и средний чек заказа   
3           Изменить структура категорий, что увеличит конверсию, т.к. пользователи быстрее найдут нужный товар   
4                              Изменить цвет фона главной страницы, чтобы увеличить вовлеченность пользователей   
5                      Добавить страницу отзывов клиентов о магазине, что позволит увеличить количество заказов   
6          Показать на главной странице баннеры с актуальными акциями и распродажами, чтобы увеличить конверсию   
7              Добавить форму подписки на все основные страницы, чтобы собрать базу клиентов для email-рассылок   
8                                                       Запустить акцию, дающую скидку на товар в день рождения   

   Reach  Impact  Confidence  Efforts  
0      3      10           8        6  
1      2       5           4       10  
2      8       3           7        3  
3      8       3           3        8  
4      3       1           1        1  
5      3       2           2        3  
6      5       3           8        3  
7     10       7           8        5  
8      1       9           9        5

In [3]:
hypothesis.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Hypothesis  9 non-null      object
 1   Reach       9 non-null      int64 
 2   Impact      9 non-null      int64 
 3   Confidence  9 non-null      int64 
 4   Efforts     9 non-null      int64 
dtypes: int64(4), object(1)
memory usage: 488.0+ bytes


In [4]:
hypothesis.columns = [x.lower() for x in hypothesis.columns.values]

#### Вывод

Описание данных
 - Hypothesis - краткое описание гипотезы
 - Reach - охват пользователей по 10-балльной шкале
 - Impact - влияние на пользователей по 10-балльной шкале
 - Confidence - уверенность в гипотезе по 10-балльной шкале
 - Efforts - затраты ресурсов на проверку гипотезы по 10-балльной шкале. Чем больше значение Efforts, тем дороже проверка          гипотезы
   

### Датасет orders

In [5]:
orders = pd.read_csv('/datasets/orders.csv', sep = ',')
orders.sample(10)

transactionId   visitorId        date  revenue group
984     3217433767  2637462286  2019-08-10      100     B
926      884109530  1909539331  2019-08-30     3430     A
742     1113045957  3323129950  2019-08-09    14490     A
849      373945924  3785537730  2019-08-05     2290     B
308     2669609377  3511496907  2019-08-19     1900     A
737     1061639574  2195309755  2019-08-09     2960     B
721      390467479  2546716293  2019-08-05     1805     A
536     3925488023  3984495233  2019-08-04    20219     A
236     3422531330  2230225652  2019-08-23      660     B
885      335158381   947532470  2019-08-05     5280     A

In [6]:
orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1197 entries, 0 to 1196
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   transactionId  1197 non-null   int64 
 1   visitorId      1197 non-null   int64 
 2   date           1197 non-null   object
 3   revenue        1197 non-null   int64 
 4   group          1197 non-null   object
dtypes: int64(3), object(2)
memory usage: 46.9+ KB


In [7]:
orders['date'] = orders['date'].map(
    lambda x: dt.datetime.strptime(x, '%Y-%m-%d')
)

In [8]:
orders['transactionId'].value_counts()

1658608862    1
2211880073    1
3913205113    1
437656952     1
3592105335    1
             ..
2872509092    1
2013919907    1
3154167297    1
2960804513    1
4281129020    1
Name: transactionId, Length: 1197, dtype: int64

In [9]:
orders['visitorId'].value_counts()

4256040402    11
2458001652    11
2378935119     9
2038680547     8
199603092      5
              ..
2406916810     1
1255412427     1
3866069709     1
54447517       1
3013339133     1
Name: visitorId, Length: 1031, dtype: int64

In [10]:
orders['date'].describe()

count                    1197
unique                     31
top       2019-08-05 00:00:00
freq                       48
first     2019-08-01 00:00:00
last      2019-08-31 00:00:00
Name: date, dtype: object

In [11]:
orders['group'].value_counts()

B    640
A    557
Name: group, dtype: int64

#### Вывод

Описание данных
 - transactionId - идентификатор заказа
 - visitirId - идентификатор пользователя, совершившего заказ
 - date - дата, когда был совершен заказ
 - revenue - выручка заказа
 - group - группа А/В теста, в которую попал заказ

Поменяли формат в поле дата. Всего у нас 1197 строк с заказами. Максимальное количество заказов одним пользователем 11. Пользователи поделены на две неравные группы А-557 человек, В - 640 человек это почти на 15% больше.

### Датасет visitors

In [12]:
visitors = pd.read_csv('/datasets/visitors.csv', sep = ',')
visitors.sample(10)

date group  visitors
27  2019-08-28     A       594
4   2019-08-05     A       756
55  2019-08-25     B       679
31  2019-08-01     B       713
5   2019-08-06     A       667
14  2019-08-15     A       628
39  2019-08-09     B       610
48  2019-08-18     B       737
11  2019-08-12     A       595
10  2019-08-11     A       718

In [13]:
visitors.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62 entries, 0 to 61
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   date      62 non-null     object
 1   group     62 non-null     object
 2   visitors  62 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 1.6+ KB


In [14]:
visitors['date'] = visitors['date'].map(
    lambda x: dt.datetime.strptime(x, '%Y-%m-%d')
)

In [15]:
visitors['date'].describe()

count                      62
unique                     31
top       2019-08-27 00:00:00
freq                        2
first     2019-08-01 00:00:00
last      2019-08-31 00:00:00
Name: date, dtype: object

In [16]:
visitors['group'].value_counts()

B    31
A    31
Name: group, dtype: int64

In [17]:
visitors['visitors'].describe()

count     62.000000
mean     607.290323
std      114.400560
min      361.000000
25%      534.000000
50%      624.500000
75%      710.500000
max      770.000000
Name: visitors, dtype: float64

In [18]:
gr = orders.pivot_table(index='visitorId', values='group', aggfunc='nunique')
gr.columns = ['group_count', 'del_col']
gr.reset_index()
gr.query('group_count > 1').count()

ValueError: Length mismatch: Expected axis has 1 elements, new values have 2 elements

#### Вывод

Описание данных
 - date - дата
 - group - группа А/В теста
 - visitors - количество пользователей в указанную дату в указанной группе А/В теста

Поменяли формат времени в поле дата. Данные собранные в датасете за месяц. У нас две группы пользователей. 58 пользователей оказались сразу в двух группах. Это 5% процентов пользователей, удалять их не будем так как это может повлиять на показатели платящих-неплатящих пользователей.

### Вывод

У нас три датасета. В первом список из 9 гипотез, которые нужно приоритизировать изходя из параметров данных в датасете.В двух других данные о пользователях и их заказах при проведение А/В теста. На основании этих данных нам нужно сделать анализ проведенного теста. Для удобства привели названия столбцов к нижнему регистру. Выяснили, что 58 пользователей попали сразу в две группы. Для анализа у нас данные о заказах за период с 1-08-2019 по 31-08-2019. Пользователи поделены на две неравные группы.

## Приоритизация гипотез

### Применить  фреймворк ICE для приоритизации гипотез.Отсортировать по убыванию приоритета.

In [ ]:
hypothesis['ice'] = ((hypothesis['impact'] * hypothesis['confidence']) / hypothesis['efforts']).round(2)

hypothesis[['hypothesis', 'ice']].sort_values(by='ice', ascending=False)


### Вывод

Применили способ приоритизации гипотез ICE. После расчета можно выделить три самые перспективные гипотезы:
- Запустить акцию, дающую скидку на товар в день рождения
- Добавить два новых канала привлечения трафика, что позволит привлекать на 30% больше пользователей
- Добавить форму подписки на все основные страницы, чтобы собрать базу клиентов для email-рассылок

### Применить  фреймворк RICE для приоритизации гипотез.Отсортировать по убыванию приоритета.

In [ ]:
hypothesis['rice'] = ((hypothesis['reach'] * hypothesis['impact'] * hypothesis['confidence']) / hypothesis['efforts']).round(2)

hypothesis[['hypothesis', 'rice']].sort_values(by='rice', ascending=False)

### Вывод

Применили способ приоритизации гипотез RICE. После расчета можно выделить три самые перспективные гипотезы:
   - Добавить форму подписки на все основные страницы, чтобы собрать базу клиентов для email-рассылок
   - Добавить блоки рекомендаций товаров на сайт интернет магазина, чтобы повысить конверсию и средний чек заказа	
   - Добавить два новых канала привлечения трафика, что позволит привлекать на 30% больше пользователей

### Указать, как изменилась приоритизация гипотез при применении RICE вместо ICE

После изменения фреймворка на первое место вышла гипотезы "Добавить форму подписки на все основные страницы, чтобы собрать базу клиентов для email-рассылок". В формулу расчета RICE добавляется параметр REACH. Это показатель, количества пользователей которое затронет изменение. В этой гипотезе показатель имеет максимальные 10 балов. Поэтому гипотеза заняла первое место. Из-за этого показателя также поменяла место гипотеза "Добавить два новых канала привлечения трафика, что позволит привлекать на 30% больше пользователей", а гипотеза "Запустить акцию, дающую скидку на товар в день рождения" вообще не вошла в тройку. Так как день рождения в период акции может совпасть только у маленького числа пользователей, соответственно и показатель REACH =1.

### Вывод

Применили на практике фреймворки ICE и RICE. Увидели как показатели IMPACT, CONFIDENCE, EFFORTS и REACH влияют на приоритизацию гипотез. Поэтому в зависимости от задачи главное выбрать правильный способ приоритизации. Если нам надо увеличить показатели влияющие на удовлетворение продуктом, и для нас не главное количество пользователей вполне подходит показатель ICE. Но если нам нужно охватить как можно большую аудиторию пользователей, для расчета подойдет только RICE

## Анализ А/В теста

### Построить график кумулятивной выручки по группам

In [ ]:
orders.head()

In [ ]:
datesGroups = orders[['date', 'group']].drop_duplicates()

In [ ]:
datesGroups.head()

In [ ]:
# получаем агрегированные кумулятивные по дням данные о заказах 
ordersAggregated = datesGroups.apply(
    lambda x: orders[
        np.logical_and(
            orders['date'] <= x['date'], orders['group'] == x['group']
        )
    ].agg(
        {
            'date': 'max',
            'group': 'max',
            'transactionId': pd.Series.nunique,
            'visitorId': pd.Series.nunique,
            'revenue': 'sum',
        }
    ),
    axis=1,
).sort_values(by=['date', 'group'])

In [ ]:
ordersAggregated.head()

In [ ]:
visitorsAggregated = datesGroups.apply(
    lambda x: visitors[
        np.logical_and(
            visitors['date'] <= x['date'], visitors['group'] == x['group']
        )
    ].agg({'date': 'max', 'group': 'max', 'visitors': 'sum'}),
    axis=1,
).sort_values(by=['date', 'group'])

In [ ]:
visitorsAggregated.head()

In [ ]:
# объединяем кумулятивные данные в одной таблице и присваиваем ее столбцам понятные названия
cumulativeData = ordersAggregated.merge(
    visitorsAggregated, left_on=['date', 'group'], right_on=['date', 'group']
)
cumulativeData.columns = [
    'date',
    'group',
    'orders',
    'buyers',
    'revenue',
    'visitors',
]

In [ ]:
cumulativeData.head()

In [ ]:
cumulativeRevenueA = cumulativeData[cumulativeData['group']=='A'][['date','revenue', 'orders']]
cumulativeRevenueA.head()

In [ ]:
# датафрейм с кумулятивным количеством заказов и кумулятивной выручкой по дням в группе B
cumulativeRevenueB = cumulativeData[cumulativeData['group']=='B'][['date','revenue', 'orders']]
cumulativeRevenueB.head()

In [ ]:
# Строим график выручки группы А
plt.plot(cumulativeRevenueA['date'], cumulativeRevenueA['revenue'], label='A')

# Строим график выручки группы B
plt.plot(cumulativeRevenueB['date'], cumulativeRevenueB['revenue'], label='B')
plt.xticks(rotation=45)
plt.legend() 
plt.title('Кумулятивная выручка по группам')
plt.xlabel('Дата, день')
plt.ylabel('Выручка, руб')
plt.show()

#### Вывод

Выручка по группа была на одном уровне только первые 5 дней. Потом выручка группы В стала рости, а 17-08-2019 был большой скачок, после которого группа вырвалась сильно вперед. Не стоит забывать, что в группе В на 15%, больше пользователей и это дает большое преимущество в выручке. Надо выяснить из-за чего был большой скачок.

### Построить график кумулятивного среднего чека по группам

In [ ]:
plt.plot(cumulativeRevenueA['date'], cumulativeRevenueA['revenue']/cumulativeRevenueA['orders'], label='A')
plt.plot(cumulativeRevenueB['date'], cumulativeRevenueB['revenue']/cumulativeRevenueB['orders'], label='B')
plt.xticks(rotation=45)
plt.title('Кумулятивный средний чек по группам')
plt.xlabel('Дата, день')
plt.ylabel('Средний чек, руб')
plt.legend() 
plt.show()

#### Вывод

На графике видно несколько скачков по среднему чеку. для группы А сначала было снижение, но 8-9-10 августа было сразу два скачка, после которых средний чек был более стабилен. В группе В большой скачок 17-18 августа, показатели взлетели за 1000 и потом стали плавно снижаться. Это мог быть большой или очень дорогой заказ.

### Построить график относительного изменения кумулятивного среднего чека группы В к группе А

In [ ]:
mergedCumulativeRevenue = cumulativeRevenueA.merge(
    cumulativeRevenueB, left_on='date', right_on='date', how='left', suffixes=['A', 'B'])

In [ ]:
mergedCumulativeRevenue.head()

In [ ]:
# cтроим отношение средних чеков
plt.plot(mergedCumulativeRevenue['date'], (mergedCumulativeRevenue['revenueB']/mergedCumulativeRevenue['ordersB'])/(mergedCumulativeRevenue['revenueA']/mergedCumulativeRevenue['ordersA'])-1)
# добавляем ось X
plt.axhline(y=0, color='black', linestyle='--') 
plt.xticks(rotation=45)
plt.title('Относительное изменение кумулятивного среднего чека группы B к группе A')
plt.xlabel('Дата, день')
plt.ylabel('Отношение средних чеков')
plt.show()

#### Вывод

В нескольких точках графика опять скачки. Причем здесь есть и провал в минус и опять большой скачок 17-18 числа, после которого наблюдается снижение чека. 

### Построить график кумулятивной конверсии по группам

In [ ]:
# считаем кумулятивную конверсию
cumulativeData['conversion'] = cumulativeData['orders']/cumulativeData['visitors']
cumulativeData

In [ ]:
# отделяем данные по группе A
cumulativeDataA = cumulativeData[cumulativeData['group']=='A']
 
# отделяем данные по группе B
cumulativeDataB = cumulativeData[cumulativeData['group']=='B']

In [ ]:
# строим графики
plt.plot(cumulativeDataA['date'], cumulativeDataA['conversion'], label='A')
plt.plot(cumulativeDataB['date'], cumulativeDataB['conversion'], label='B')
plt.legend()
plt.xticks(rotation=45)
plt.title("Кумулятивная конверсия по группам")
plt.xlabel('Дата, день')
plt.ylabel('Конверсия')
 
# задаем масштаб осей
plt.axis(['2019-08-01', '2019-08-31', 0.025, 0.04])

plt.show()

#### Вывод

Сначала в группе А был скачок конверсии, потом значения упали и колебания остались на уровне 0,03. В группе В тоже вначале был большой скачок конверсии, но затем показатели выровнялись и конверсия осталась на уровне 0,034. В принципе колебания не большие, все изменения происходят в пределах 1%.  

### Построить график относительного изменения кумулятивной конверсии группы В к группе А

In [ ]:
mergedCumulativeConversions = cumulativeDataA[['date','conversion']].merge(
    cumulativeDataB[['date','conversion']], left_on='date', right_on='date', how='left', suffixes=['A', 'B'])

mergedCumulativeConversions.head()

In [ ]:
plt.figure(figsize=(8,6))
plt.plot(
    mergedCumulativeConversions['date'], 
    mergedCumulativeConversions['conversionB']/mergedCumulativeConversions['conversionA']-1, 
    label="Относительный прирост конверсии группы B относительно группы A"
)
plt.legend()
plt.xlabel('Дата, день')
plt.ylabel('Значение прироста')

plt.xticks(rotation=45)

#Добавим пунктирную линию оси X чёрного цвета ('black') со значением оси Y = 0 методом plt.axhline(). 
plt.axhline(y=0, color='black', linestyle='--')
#И ещё одну линию оси X серого цвета ('grey') со значением оси Y = 0.2.
plt.axhline(y=0.2, color='grey', linestyle='--')

plt.axis(["2019-08-01", '2019-08-31', -0.2, 0.3])

plt.show()

#### Вывод

Вначале был большой скачок на 15% вверх, а потом и вниз конверсии группы В. После 3 августа конверсия стала постоянно расти до 17 августа, и только потом начался стабильный плавный спад. В конце месяца конверсия установилась на уровне 10-13%

### Построить точечный график количества заказов по пользователям

In [ ]:
ordersByUsers = (
    orders.drop(['group', 'revenue', 'date'], axis=1)
    .groupby('visitorId', as_index=False)
    .agg({'transactionId': pd.Series.nunique})
)

ordersByUsers.columns = ['userId', 'orders']

ordersByUsers.sort_values(by='orders', ascending=False).head(10)

In [ ]:
# серия из чисел от 0 до количества наблюдений в ordersByUsers
x_values = pd.Series(range(0, len(ordersByUsers)))

plt.scatter(x_values, ordersByUsers['orders']) 

plt.title('Распределение числа заказов по пользователям')
plt.xlabel('Число пользователей, чел')
plt.ylabel('Число заказов, шт')
plt.show()

#### Вывод

Большинство пользователей делает только один заказ. Но на самом деле достаточно большое количество пользователей делает два заказа. По графику также видно, что есть пользователи сделавшие 3-5 заказов. И даже несколько пользователей, которых можно прчислить к аномальным с их 8-11 заказами.

### Посчитать 95-й и 99-й перцентили количества заказов на пользователя. Выбрать границу для определения аномальных заказов

In [ ]:
# Метод np.percentile('column', [percentile1, percentile2, percentile3]) находит перцентили.
print(np.percentile(ordersByUsers['orders'], [95, 99]))

#### Вывод

Не больше 5% пользователей делали 2 заказа, и не более 1% делали 4 и более заказа. Разумно выбрать 4 заказа на одного пользователя за нижнию границу, и отсеять аномальных пользователей по ней.

### Построить точечный график стоимости заказов.

In [ ]:
x_values = pd.Series(range(0, len(orders['revenue'])))

plt.scatter(x_values, orders['revenue']) 

plt.title('Распределение стоимости заказов')
plt.xlabel('Число пользователей, чел')
plt.ylabel('Стоимость заказов, руб')
plt.axis([0, 1200, 0, 130000])

plt.show()

#### Вывод

Большинство заказов находятся в диапозоне до 20000 рублей. Достаточно большое количество заказов приходится на диапазон 20000-50000. А вот более дорогие заказы встречаются гораздо реже.  

### Посчитать 95-й и 99-й перцентили стоимости заказов. Выбрать границу для определения аномальных заказов

In [ ]:
print(np.percentile(orders['revenue'], [95,97.5, 99])) 

#### Вывод

Не более 5% пользователей делают заказ дороже 28000, и не более 1% пользователей дороже 58000. Границу аномальных заказов можно установить на 28000. В этот диапазон входят достаточно большое количество дорогих заказов, но уже исключаются крупные заказы влияющие на расчеты.

### Посчитать статистическую значимость различий в конверсии между группами по "сырым" данным.

In [ ]:
# получаем агрегированные кумулятивные по дням данные о посетителях группы А
visitorsADaily = visitors[visitors['group'] == 'A'][['date', 'visitors']]
visitorsADaily.columns = ['date', 'visitorsPerDateA']
visitorsADaily.head()

In [ ]:
visitorsACummulative = visitorsADaily.apply(
    lambda x: visitorsADaily[visitorsADaily['date'] <= x['date']].agg(
        {'date': 'max', 'visitorsPerDateA': 'sum'}
    ),
    axis=1,
)
visitorsACummulative.columns = ['date', 'visitorsCummulativeA']
visitorsACummulative.head()

In [ ]:
# получаем агрегированные кумулятивные по дням данные о посетителях группы В
visitorsBDaily = visitors[visitors['group'] == 'B'][['date', 'visitors']]
visitorsBDaily.columns = ['date', 'visitorsPerDateB']

In [ ]:
visitorsBCummulative = visitorsBDaily.apply(
    lambda x: visitorsBDaily[visitorsBDaily['date'] <= x['date']].agg(
        {'date': 'max', 'visitorsPerDateB': 'sum'}
    ),
    axis=1,
)
visitorsBCummulative.columns = ['date', 'visitorsCummulativeB']
visitorsBCummulative.head()

In [ ]:
# получаем агрегированные по дням данные о заказах и выручке группы А
ordersADaily = (
    orders[orders['group'] == 'A'][['date', 'transactionId', 'visitorId', 'revenue']]
    .groupby('date', as_index=False)
    .agg({'transactionId': pd.Series.nunique, 'revenue': 'sum'})
)
ordersADaily.columns = ['date', 'ordersPerDateA', 'revenuePerDateA']


In [ ]:
ordersADaily.head()

In [ ]:
# получаем кумулятивные по дням данные о заказах и выручке группы А
ordersACummulative = ordersADaily.apply(
    lambda x: ordersADaily[ordersADaily['date'] <= x['date']].agg(
        {'date': 'max', 'ordersPerDateA': 'sum', 'revenuePerDateA': 'sum'}
    ),
    axis=1,
).sort_values(by=['date'])

ordersACummulative.columns = [
    'date',
    'ordersCummulativeA',
    'revenueCummulativeA',
]

ordersACummulative.head()

In [ ]:
# получаем агрегированные по дням данные о заказах и выручке группы В
ordersBDaily = (
    orders[orders['group'] == 'B'][['date', 'transactionId', 'visitorId', 'revenue']]
    .groupby('date', as_index=False)
    .agg({'visitorId': pd.Series.nunique, 'revenue': 'sum'})
)
ordersBDaily.columns = ['date', 'ordersPerDateB', 'revenuePerDateB']
ordersBDaily.head()

In [ ]:
# получаем кумулятивные по дням данные о заказах и выручке группы В
ordersBCummulative = ordersBDaily.apply(
    lambda x: ordersBDaily[ordersBDaily['date'] <= x['date']].agg(
        {'date': 'max', 'ordersPerDateB': 'sum', 'revenuePerDateB': 'sum'}
    ),
    axis=1,
).sort_values(by=['date'])

ordersBCummulative.columns = [
    'date',
    'ordersCummulativeB',
    'revenueCummulativeB',
]
ordersBCummulative.head()

In [ ]:
# Объединяем по дате все полученные кумулятивные данные в один датафрейм
data = (
    ordersADaily.merge(
        ordersBDaily, left_on='date', right_on='date', how='left'
    )
    .merge(ordersACummulative, left_on='date', right_on='date', how='left')
    .merge(ordersBCummulative, left_on='date', right_on='date', how='left')
    .merge(visitorsADaily, left_on='date', right_on='date', how='left')
    .merge(visitorsBDaily, left_on='date', right_on='date', how='left')
    .merge(visitorsACummulative, left_on='date', right_on='date', how='left')
    .merge(visitorsBCummulative, left_on='date', right_on='date', how='left')
)

data.head(5)

In [ ]:
ordersByUsersA = (
    orders[orders['group'] == 'A']
    .groupby('visitorId', as_index=False)
    .agg({'transactionId': pd.Series.nunique})
)
ordersByUsersA.columns = ['userId', 'orders']
ordersByUsersA.head()

In [ ]:
ordersByUsersB = (
    orders[orders['group'] == 'B']
    .groupby('visitorId', as_index=False)
    .agg({'transactionId': pd.Series.nunique})
)
ordersByUsersB.columns = ['userId', 'orders'] 
ordersByUsersB.head()

In [ ]:
#в sampleA сохраняем выборку, где каждый элемент — число заказов определённого пользователя, в том числе ноль. 
# число элементов sampleA — это количество пользователей, сумма всех элементов — количество заказов.
sampleA = pd.concat(
    [
        ordersByUsersA['orders'],
        pd.Series(
            0,
            index=np.arange(
                data['visitorsPerDateA'].sum() - len(ordersByUsersA['orders'])
            ),
            name='orders',
        ),
    ],
    axis=0,
)
sampleA.count()

In [ ]:
sampleB = pd.concat(
    [
        ordersByUsersB['orders'],
        pd.Series(
            0,
            index=np.arange(
                data['visitorsPerDateB'].sum() - len(ordersByUsersB['orders'])
            ),
            name='orders',
        ),
    ],
    axis=0,
)
sampleB.count()

Получили Series с 18916 значений.

Сформулируем гипотезы

Н0 - различия в конверсиях между группами по «сырым» данным статистически не значимы

Н1 - различия в конверсиях между группами по «сырым» данным статистически значимы

Пороговое значение alpha примем за 5%.

У реализации метода Манн-Уитни в python есть одна особенность: его базовые настройки в python не предполагают проверку двухсторонней гипотезы, а это не соответствует гипотезам, проверяемым в проекте. Для корректной работы метода необходимо задать его параметр alternative = 'two-sided'



In [ ]:
p_val_conversion = (sps.mannwhitneyu(sampleA, sampleB,alternative = 'two-sided')[1]).round(5)

print("P-value по конверсии по сырым данным: {0:.3f}".format(p_val_conversion))

In [ ]:
relative_increase_conversion = sampleB.mean() / sampleA.mean() - 1
print("Относительный прирост конверсии группы B: {0:.3f}".format(relative_increase_conversion))

In [ ]:
alpha = .05

if (p_val_conversion < alpha):
    print("Отвергаем нулевую гипотезу")
else:
    print("Не получилось отвергнуть нулевую гипотезу")

#### Вывод

Отвергаем нулевую гипотезу и принимаем H1. P-value меньше 5%, а это значит, что различия в конверсиях между группами по «сырым» данным статистически значимы.Прирост конверсии группы В 13,8%. 

### Посчитать статистическую значимость различий в среднем чеке заказа между группами по "сырым" данным.

Сформулируем гипотезы

Н0 - различия в среднем чеке заказа между группами по «сырым» данным статистически не значимы

Н1 - различия в среднем чеке заказа между группами по «сырым» данным статистически значимы

Пороговое значение alpha примем за 5%.

Гипотеза у нас двусторонняя (есть нулевая и альтернативная), поэтому добавим в них параметр alternative = 'two-sided'.

Чтобы рассчитать статистическую значимость различий в среднем чеке между сегментами, передадим критерию mannwhitneyu() данные о выручке с заказов.

In [ ]:
revenue_A = orders[orders['group']=='A']['revenue']

In [ ]:
revenue_B = orders[orders['group']=='B']['revenue']

In [ ]:
p_val_check = (sps.mannwhitneyu(revenue_A, revenue_B,alternative = 'two-sided')[1]).round(3)

print('P-value по среднему чеку до очистки данных: {}'.format(p_val_check))

In [ ]:
relative_increase_check = revenue_B.mean() / revenue_A.mean() - 1

print('Относительный прирост среднего чека группы B к А по сырым данным: {:.2f}'.format(relative_increase_check))

In [ ]:
alpha = .05

if (p_val_check < alpha):
    print("Отвергаем нулевую гипотезу")
else:
    print("Не получилось отвергнуть нулевую гипотезу")
    

#### Вывод

Подтвердили нулевую гипотезу - различия в среднем чеке заказа между группами по «сырым» данным статистически не значимы. P value равно 0,729 что значительно превышает наш порог. Прирост среднего чека группы В 

### Посчитать статистическую значимость различий в конверсии между группами по "очищенным" данным 

Сделаем срезы пользователей с числом заказов 4 и более — usersWithManyOrders и пользователей, совершивших заказы дороже 28 000 — usersWithExpensiveOrders.

In [ ]:
usersWithManyOrders = pd.concat(
    [
        ordersByUsersA[ordersByUsersA['orders'] > 4]['userId'],
        ordersByUsersB[ordersByUsersB['orders'] > 4]['userId'],
    ],
    axis=0,
)

In [ ]:
usersWithExpensiveOrders = orders[orders['revenue'] > 30000]['visitorId']

In [ ]:
abnormalUsers = (
    pd.concat([usersWithManyOrders, usersWithExpensiveOrders], axis=0)
    .drop_duplicates()
    .sort_values()
)
abnormalUsers.head(5)

In [ ]:
abnormalUsers.shape

Получили множество 53 аномальных пользователя. После их удаления можно считать статистическую значимость.

Сформулируем гипотезы

Н0 - различия в конверсии между группами по «очищенным» данным статистически не значимы

Н1 - различия в конверсии между группами по «очищенным» данным статистически значимы

Пороговое значение alpha примем за 5%.

Создадим переменные sampleAFiltered и sampleBFiltered, в которых сохраним очищенные данные о заказах — не включая аномальных пользователей.

In [ ]:
sampleAFiltered = pd.concat(
    [
        ordersByUsersA[
            np.logical_not(ordersByUsersA['userId'].isin(abnormalUsers))
        ]['orders'],
        pd.Series(
            0,
            index=np.arange(
                data['visitorsPerDateA'].sum() - len(ordersByUsersA['orders'])
            ),
            name='orders',
        ),
    ],
    axis=0,
)

In [ ]:
sampleBFiltered = pd.concat(
    [
        ordersByUsersB[
            np.logical_not(ordersByUsersB['userId'].isin(abnormalUsers))
        ]['orders'],
        pd.Series(
            0,
            index=np.arange(
                data['visitorsPerDateB'].sum() - len(ordersByUsersB['orders'])
            ),
            name='orders',
        ),
    ],
    axis=0,
) 

In [ ]:
p_val_conversion_clean = (sps.mannwhitneyu(sampleAFiltered, sampleBFiltered, alternative = 'two-sided')[1]).round(5)

print('P-value по конверсии после очистки данных: {}'.format(p_val_conversion_clean))

In [ ]:
relative_increase_conversion_clean = sampleBFiltered.mean()/sampleAFiltered.mean()-1

print('Относительный прирост конверсии группы B к конверсии группы А после очистки данных: {:.2f}'.format(relative_increase_conversion_clean))

In [ ]:
alpha = .05

if (p_val_conversion_clean < alpha):
    print("Отвергаем нулевую гипотезу")
else:
    print("Не получилось отвергнуть нулевую гипотезу")

#### Вывод

P value равен 0,01 значит мы отвергаем Нулевую гипотезу и принимаем альтернативную. Различия в конверсии между группами по «очищенным» данным статистически значимы. Прирост конверсии группы В по очищенным данным составляет 14%.

### Посчитать статистическую значимость различий в среднем чеке заказа между группами по "очищенным" данным.

Посчитаем статистическую значимость различия средних чеков после удаления аномальных пользователей.

Сформулируем гипотезы

Н0 - различия в среднем чеке заказа между группами по «очищенным» данным статистически не значимы

Н1 - различия в среднем чеке заказа между группами по «очищенным» данным статистически значимы

Пороговое значение alpha примем за 5%.

Гипотеза у нас двусторонняя (есть нулевая и альтернативная), поэтому добавим в них параметр alternative = 'two-sided'.

In [ ]:
#отфильтруем аномальных пользователей
sampleAFiltered_revenue = orders[
                np.logical_and(
                    orders['group'] == 'A',
                    np.logical_not(orders['visitorId'].isin(abnormalUsers)),
                )
            ]['revenue']

sampleBFiltered_revenue = orders[
                np.logical_and(
                    orders['group'] == 'B',
                    np.logical_not(orders['visitorId'].isin(abnormalUsers)),
                )
            ]['revenue']

In [ ]:
p_val_avg_check_clean = (sps.mannwhitneyu(
    sampleAFiltered_revenue, 
    sampleBFiltered_revenue, 
    alternative = 'two-sided')[1]).round(5)

print('P-value по среднему чеку после очистки данных: {:.3f}'.format(p_val_avg_check_clean))

In [ ]:
relative_increase_avg_check_clean = sampleBFiltered_revenue.mean()/sampleAFiltered_revenue.mean()-1

print('Относительный прирост среднего чека группы B к среднему чеку группы А после очистки данных: {:.2f}'.format(relative_increase_avg_check_clean))

In [ ]:
alpha = .05

if (p_val_avg_check_clean < alpha):
    print("Отвергаем нулевую гипотезу")
else:
    print("Не получилось отвергнуть нулевую гипотезу")

#### Вывод

Мы подтвердили нулевую гипотезу, что различия в среднем чеке заказа между группами по «очищенным» данным статистически не значимы. Относительный прирост в чеке по очищенным данным упал с 0.26 до -0.02. 

### Принять решения по результатам теста

Примите решение по результатам теста и объясните его. Варианты решений: 1. Остановить тест, зафиксировать победу одной из групп. 2. Остановить тест, зафиксировать отсутствие различий между группами. 3. Продолжить тест.

Опираясь на полученные результаты и несмотря на то, что по среднему чеку статистической значимости не обнаружено, нужно остановить тест и зафиксировать победу группы В. Группа B статистически значимо лучше группы A по метрике конверсии, но различий в средних чеках нет. Так как задача стоит в эффективности гипотезы по увеличению выручки, нужно смотреть на конверсию. Выручка не всегда покажет положительную или отрицательную динамику, точнее сложно понять по выручке, без учета числа покупателей.

Коверсия влияет на выручку, и поэтому раз конверсия в случае группы В значимо увеличилась, то и выручка увеличилась тоже. Даже с учетом того, что средний чек остался без изменений.

#### Вывод

Опираясь на результаты тестов можно остановить тест. Мы проверяли основную гипотезу по увеличению выручки. По графикам у нас показатели среднего чека в группе В рос, по отношению к среднему чеку группы А. При проверке гипотезы по увелечению среднего чека, мы не получили статистически значимых результатов, что средний чек увеличился.  Посчитав статистическую значимость различий в конверсии мы убедились, что различия в показателях группы В  статистически значимы. По графикам кумулятивной конверсии тоже виден рост с небольшими коллебаниями, но конверсия растет. Следовательно мы получили увеличение выручки в группе В. 

### Вывод

- Провели анализ кумулятивной выручки по групам, подтвердили ее рост в обеих группах.
- Проанализировали средний чек по группам, обнаружили несколько скачков при отстутствии постоянного стабильного роста.
- Изменение кумулятивного среднего чека группы В к группе А нестабильно. По графику наблюдаются большие скачки и вверх и провалы до минуса
- Кумулятивная конверсия в группе В выше, чем в группе. Но изменения колеблются в пределах 1%
- По графику кумулятивной конверсии группы В к группе А наблюдается стабильный рост. Был небольшой спад после 20% прироста, но потом конверсия стала опять рости.
- По графику количества заказов наблюдается приоритет заказов 2-4. Отчетливо видно, что количество заказов больше 5 редкие, аномальные. 
- Не больше 5% пользователей делали 2 заказа, и не более 1% делали 4 и более заказа. Разумно выбрать 4 заказа на одного пользователя за нижнию границу, и отсеять аномальных пользователей по ней.
- По точечному графику видно, что большая частьзаказов сфокусирована в пределах 20000
- Не более 5% пользователей делают заказ дороже 28000, и не более 1% пользователей дороже 58000. 
- Различия в конверсиях между группами по «сырым» данным статистически значимы.
- Различия в среднем чеке заказа между группами по «сырым» данным статистически не значимы
- Различия в конверсии между группами по «очищенным» данным статистически значимы
- Различия в среднем чеке заказа между группами по «очищенным» данным статистически не значимы.
- По всем имеющимся данным принято решение тест остановить. Группа В победила.


## Общий вывод

Получили три датасета с данными для анализа крупного интернет магазина. Цель увеличение выручки. На входе проверили все данные. Поменяли где необходимо формат, привели к нижнему регистру столбцы.

Часть 1.
- У нас 9 гипотез для приоритизации.

По фреймворку ICE( влияние, уверенность, усилия)  три приоритетные гипотезы :
   - Запустить акцию, дающую скидку на товар в день рождения
   - Добавить два новых канала привлечения трафика, что позволит привлекать на 30% больше пользователей
   - Добавить форму подписки на все основные страницы, чтобы собрать базу клиентов для email-рассылок
   
По фреймворку RICE добавляется показатель REACH (скольких пользователей затронет изменения), который меняет приоритеты:
   - Добавить форму подписки на все основные страницы, чтобы собрать базу клиентов для email-рассылок
   - Добавить блоки рекомендаций товаров на сайт интернет магазина, чтобы повысить конверсию и средний чек заказа
   - Добавить два новых канала привлечения трафика, что позволит привлекать на 30% больше пользователей
   
Поэтому в зависимости от задачи главное выбрать правильный способ приоритизации. Если нам надо увеличить показатели влияющие на удовлетворение продуктом, и для нас не главное количество пользователей вполне подходит показатель ICE. Но если нам нужно охватить как можно большую аудиторию пользователей, для расчета подойдет только RICE   

Часть 2.

На входе получили данные из двух датасетов с данными о пользователях, группах к которым они относятся и их заказах. У нас 1197 пользователей поделенных на 640-группа В и 557-группа А. 58 пользователей входит в две группы. Удалять их не стали, их порядка 5% поэтому влиять на анализ они не должны. Данные по заказам у нас за промежуток с 1-08-2019 по 31-08-2019. 

Проанализированы кумулятивные метрики:
  - Подтвердили рост выручки в обеих группах.
  - Обнаружили несколько скачков по увеличению среднего чека, при отстутствии постоянного стабильного роста.
  - Изменение кумулятивного среднего чека группы В к группе А нестабильно. По графику наблюдаются большие скачки и вверх и         провалы до минуса.
  - Конверсия в группе В выше, чем в группе. Но изменения колеблются в пределах 1%
  - Наблюдается стабильный рост конверсии группы В к группе А. Был небольшой спад после 20% прироста, но потом конверсия стала     опять рости.
  
Проанализированы аномалии по количеству заказов и стоимости заказов:
   - Большая часть пользователей делает 2-4. Есть редкие заказы от 5 и выше, некоторые пользователи совершали до 11 заказов.
   - Не больше 5% пользователей делали 2 заказа, и не более 1% делали 4 и более заказа. Что говорит о том, что более 4 заказов      это аномальные данные, которые могут влиять на расчет показателей и гипотезы.
   - Большинство заказов не превышают 20000.
   - Не более 5% пользователей делают заказ дороже 28000, и не более 1% пользователей дороже 58000. Заказы превышающие эти          цифры редкие и их можно отнести к аномальным и для расчета гипотез их надо исключать.
   
Изучена статистическая сначимость различий в среднем чеке и конверсии по "сырым" и очищенным данным.
   - В конверсии между группами по "сырым" данным есть статистически значимые различия.
   - Относительный прирост конверсии группы B к конверсии группы А равен 13%.
   - Различия в конверсии между группами по «очищенным» данным статистически значимы.
   - Статистически значимых различий в среднем чеке заказа между группами по «сырым» данным нет. 
   - Различия в среднем чеке заказа между группами по «очищенным» данным статистически не значимы.
   
   
Опираясь на результаты анализа принято решение остановить тест. Группа В победила.Мы проверяли основную гипотезу по увеличению выручки. По графикам у нас показатели среднего чека в группе В росли, по отношению к среднему чеку группы А. При проверке гипотезы по увелечению среднего чека, мы не получили статистически значимых результатов, что средний чек увеличился. Посчитав статистическую значимость различий в конверсии мы убедились, что различия в показателях группы В статистически значимы. По графикам кумулятивной конверсии тоже виден рост с небольшими коллебаниями, но конверсия растет. Следовательно мы получили увеличение выручки в группе В.     